cmd/access/node_builder/access_node_builder.go

```diff
@@ -579,12 +580,19 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 			return stateSyncCommands.NewReadExecutionDataCommand(builder.ExecutionDataStore)
 		}).
 		Module("transactions and collections storage", func(node *cmd.NodeConfig) error {
-			// TODO: needs to be wrapped with ChainedCollections module, otherwise once we switch
-			// ProtocolDB to pebble based storage, the data previously stored in badger will not be
-			// accessable.
+
 			transactions := store.NewTransactions(node.Metrics.Cache, node.ProtocolDB)
-			builder.collections = store.NewCollections(node.ProtocolDB, transactions)
+			collections := store.NewCollections(node.ProtocolDB, transactions)
+
+			if node.ProtocolDB.ID() == storage.PebbleDBID && node.DB != nil {
```

Comment:
@zhangchiqing Maybe we can have a function to decide whether to use multiple stores to replace this condition here.

BTW, great meeting today!

Reply:
The migration is one way. We only migrate from badger to pebble, never from pebble to badger. Once we migrate to pebble, if things go wrong, we will either fix the issue, or we drop the pebble database, and recover from the badger db that was untouched since the migration. 

In other words, if the data is now stored in pebble, then it means some old data might be stored in badger, and we will need to query data from both pebble and badger. That's why I just check if data is storing in pebble to decide whether to query both or not.

Reply:
Yeah, great discussion today! 🎉